surfsense_backend/app/agents/researcher/qna_agent/nodes.py

```
@@ -0,0 +1,148 @@
+    # Initialize LLM
+    llm = app_config.fast_llm_instance
+    
+    # If no documents were provided, return a message indicating this
+    if not documents or len(documents) == 0:
+        return {
+            "final_answer": "I don't have any relevant documents in your personal knowledge base to answer this question. Please try asking about topics covered in your saved content, or add more documents to your knowledge base."
+        }
+    
+    # Prepare documents for citation formatting
+    formatted_documents = []
+    for i, doc in enumerate(documents):
```

Ruff (0.11.9)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)